File: Practica de Programacion/Practica 4 Composicion y Agregacion/Ejercicio 3/3 avion.java
import java.util.ArrayList;

class Parte {
    private String nombre;
    private int peso;
    
    public Parte(String nombre, int peso) {
        this.nombre = nombre;
        this.peso = peso;
    }
    
    public void mostrarInfo() {
        System.out.println(nombre + ": " + peso + "kg");
    }
}

class Avion {
    private String modelo;
    private String fabricante;
    private ArrayList<Parte> partes;
    
    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new ArrayList<>();
        
        partes.add(new Parte("Alas", 2));
        partes.add(new Parte("Motores", 5));
        partes.add(new Parte("Tren de aterrizaje", 3));
    }
    
    public void agregarParte(String nombre, int peso) {
        partes.add(new Parte(nombre, peso));
    }
    
    public void mostrarAvion() {
        System.out.println("\nAvión " + modelo + " de " + fabricante + ":");
        
        int totalPeso = 0;
        for (Parte parte : partes) {
            parte.mostrarInfo();
            totalPeso += parte.peso;
        }
        
        System.out.println("Total: " + partes.size() + " partes, " + totalPeso + "kg");
    }
}

public class Main {
    public static void main(String[] args) {
        Avion avion = new Avion("Boeing 747", "Boeing");
        avion.agregarParte("Puerta", 10);
        avion.mostrarAvion();
    }
}

File: Practica de Programacion/Practica 4 Composicion y Agregacion/Ejercicio 3/3 avion.py
class Parte:
    def __init__(self, nombre, peso):
        self.nombre = nombre
        self.peso = peso
        
    def mostrar_info(self):
        print(f"{self.nombre}: {self.peso}kg")
        
        
class Avion:
    def __init__(self, modelo, fabricante):
        self.modelo = modelo 
        self.fabricante = fabricante
        self.partes = [  
            Parte("Alas", 2), 
            Parte("Motores", 5), 
            Parte("Tren de aterrizaje", 3)  
        ]
        
    def agregar_parte(self, nombre, peso):  
        nueva_parte = Parte(nombre, peso)
        self.partes.append(nueva_parte)
        
    def mostrar_avion(self):
        print(f"\nAvión {self.modelo} de {self.fabricante}:")  
        for parte in self.partes: 
            parte.mostrar_info()
        total_peso = sum(p.peso for p in self.partes)
        print(f"Total: {len(self.partes)} partes, {total_peso}kg")

avion = Avion("Boeing 747", "Boeing")
avion.agregar_parte("Puerta", 10)  
avion.mostrar_avion()